### DISCUSSION TITLE
getApplicationContext from Capacitor Plugin

### DISCUSSION THREAD

**Discussion by w-mazed** [ionic-team/capacitor#5688] [Q&A] [answered]
Hi guys,

I'm tryin to code my own plugin, but when I call 

`getActivity().getApplicationContext()` or even `getActivity()`

Inside my plugin class I got 

`Attempt to invoke virtual method 'androidx.appcompat.app.AppCompatActivity com.getcapacitor.Bridge.getActivity()' on a null object reference`

Any solution for this ?

Best regards.

**Comment by robingenz** [comment]
You should be able to call `this.getActivity().getApplicationContext()` in a class that extends `com.getcapacitor.Plugin`.

Example: https://github.com/ionic-team/capacitor-plugins/blob/d97898662d0ba037e5f8448990a91de5ec6a4234/app-launcher/android/src/main/java/com/capacitorjs/plugins/applauncher/AppLauncherPlugin.java#L25

Can you share some code?

**Comment by w-mazed** [comment, marked as answer]
Thank's @robingenz, I just tried the code that you provided, but still have the same error 
```
package com.mycompany.mypackage;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "MyPackage")
public class MyPackagePlugin extends Plugin {

    private MyPackage implementation = new MyPackage(this.getActivity().getApplicationContext());

    @PluginMethod
    public void echo(PluginCall call) {
        String value = call.getString("value");

        JSObject ret = new JSObject();
        ret.put("value", implementation.echo(value));
        call.resolve(ret);
    }
}
```

**Reply by robingenz** [reply]
Try to use the `void load()` method: https://github.com/robingenz/capacitor-badge/blob/cf29ce1d68628a328076d7ed34d21a11ce6eb927/android/src/main/java/dev/robingenz/capacitor/badge/BadgePlugin.java#L17

**Reply by w-mazed** [reply]
Works like a charm, thanks and have a great day :)

**Comment by deepakrout** [comment]
I am developing a custom plugin and I am having similar issue. It seems it can't find the plugin, calls based on the error from PluginHandle.java line#104:

```
public Plugin load() throws PluginLoadException {
        if (this.instance != null) {
            return this.instance;
        }
        try {
            **this.instance = this.pluginClass.getDeclaredConstructor().newInstance();**
            return this.loadInstance(instance);
        } catch (Exception ex) {
            throw new PluginLoadException("Unable to load plugin instance. Ensure plugin is publicly accessible");
        }
    }
```
The actual error from the Exception block is:

`java.lang.NullPointerException: Attempt to invoke virtual method 'android.content.Context com.getcapacitor.Bridge.getContext()' on a null object reference.`

There are no build issues. As per the accepted answer, I have the @Override load method defined as well. I have gone through numerous stackoverflow questions, github issues, Capacitor documentation etc.; none of those fixed the issue. Without debugging the Capacitor java code, I just see "Plugin not implemented in android" message in Javascript console. 

I would really appreciate any help in this regard.

**Reply by alessandroAmedei** [reply]
Probably your "Override" load method declared in your custom plugin is throwing an Exception.
Run debug and check the message of ex variable.
